src/realtime_metrics/main.py

```diff
@@ -6,17 +6,168 @@
+    logger.info("ETA accuracy:")
+    eta_accuracy(stop_time_updates=stop_time_updates)
+
+
+def mse_accuracy(stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]):
+    # computes the accuracy of the given stop time updates using mean squared error
+
+    if len(stop_time_updates) <= 0:
+        logger.info("No samples provided!")
+        return
+
+    # compute prediction error for each stop time update
+    samples = []
+    for update in stop_time_updates:
+        trip_update = update[0]
+        stop_time_update = update[1]
+        actual_delay = get_actual_delay(trip_update=trip_update, stop_time_update=stop_time_update)
+        predicted_delay = stop_time_update.arrival_delay
+        prediction_error = actual_delay - predicted_delay
+        samples.append(prediction_error)
+
+    # compute MSE
+    sum = 0
+    for entry in samples:
+        sum += entry * entry
+    mean_squared_error = sum / len(samples)
+
+    # print MSE
+    logger.info("MSE: %s", mean_squared_error)
+
+
+def eta_accuracy(stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]):
+    # computes the accuracy of the given stop time updates using the eta bucketing approach
+
+    # initialize buckets with zero correct and incorrect samples each
+    buckets: list[list] = []
+    buckets.append([0,0])
+    buckets.append([0,0])
+    buckets.append([0,0])
+    buckets.append([0,0])
+    
+    # sort stop time updates into the correct buckets
+    for update in stop_time_updates:
+        trip_update = update[0]
+        stop_time_update = update[1]
+        actual_arrival_time = get_actual_arrival_time(trip_update=trip_update, stop_time_update=stop_time_update)
+        time_variance = actual_arrival_time - trip_update.timestamp.timestamp()
+
+        bucket_index = 0
+        upper_limit = 0
+        lower_limit = 0
+        if time_variance <= 0:
+            # remove stop time updates published after arrival
+            continue
+        elif time_variance <= 180:
+            # 0-3 min bucket
+            bucket_index = 0
+            upper_limit = 90
+            lower_limit = -30
+        elif time_variance <= 360:
+            # 3-6 min bucket
+            bucket_index = 1
+            upper_limit = 150
+            lower_limit = -60
+        elif time_variance <= 600:
+            # 6-10 min bucket
+            bucket_index = 2
+            upper_limit = 210
+            lower_limit = -60
+        elif time_variance <= 900:
+            # 10-15 min bucket
+            bucket_index = 3
+            upper_limit = 270
+            lower_limit = -90
+        else: 
+            # too far in the past
+            #print("Stop Time Update is outside the allowed interval.")
+            continue
+
+        # check if inside allowed interval
+        actual_delay = get_actual_delay(trip_update=trip_update, stop_time_update=stop_time_update)
+        predicted_delay = stop_time_update.arrival_delay
+        # TODO: fix calculation
```

Comment:
Can you explain what is wrong here in the TODO or if its too much, create an issue for it

Reply:
just forgot to remove the tag